src/Files.App/Views/Shells/BaseShellPage.cs

```diff
@@ -248,36 +248,39 @@ protected async void FilesystemViewModel_DirectoryInfoUpdated(object sender, Eve
 					? await GitHelpers.GetRepositoryHead(InstanceViewModel.GitRepositoryPath)
 					: null;
 
-			if (InstanceViewModel.GitRepositoryPath != FilesystemViewModel.GitDirectory)
+			if (FilesystemViewModel.GitDirectory is not null)
```

Comment:
I think this process should be executed even if `FilesystemViewModel.GitDirectory` is null.